internal/scout/resources/resources.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/docker/docker/client"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Option = func(config *Config)
+
+type Config struct {
+	namespace    string
+	docker       bool
+	k8sClient    *kubernetes.Clientset
+	dockerClient *client.Client
+}
+
+func WithNamespace(namespace string) Option {
+	return func(config *Config) {
+		config.namespace = namespace
+	}
+}
+
+// ResourcesK8s prints the CPU and memory resource limits and requests for all pods in the given namespace.
+func K8s(ctx context.Context, clientSet *kubernetes.Clientset, restConfig *rest.Config, opts ...Option) error {
+	cfg := &Config{
+		namespace:    "default",
+		docker:       false,
+		k8sClient:    clientSet,
+		dockerClient: nil,
+	}
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	return listPodResources(ctx, cfg)
+}
+
+func listPodResources(ctx context.Context, cfg *Config) error {
+	podInterface := cfg.k8sClient.CoreV1().Pods(cfg.namespace)
+	podList, err := podInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing pods: ")
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	defer func() {
+		_ = w.Flush()
+	}()
+
+	fmt.Fprintln(w, "CONTAINER\tCPU LIMITS\tCPU REQUESTS\tMEM LIMITS\tMEM REQUESTS\tCAPACITY")
+
+	for _, pod := range podList.Items {
+		if pod.GetNamespace() == cfg.namespace {
+			for _, container := range pod.Spec.Containers {
+				cpuLimits := container.Resources.Limits.Cpu()
+				cpuRequests := container.Resources.Requests.Cpu()
+				memLimits := container.Resources.Limits.Memory()
+				memRequests := container.Resources.Requests.Memory()
+
+				capacity, err := getPVCCapacity(ctx, cfg, container, pod)
+				if err != nil {
+					return err
+				}
+
+				fmt.Fprintf(
+					w,
+					"%s\t%s\t%s\t%s\t%s\t%s\t\n",
+					container.Name,
+					cpuLimits,
+					cpuRequests,
+					memLimits,
+					memRequests,
+					capacity,
+				)
+			}
+		}
+	}
+
+	return nil
+}
+
+func getPVCCapacity(ctx context.Context, cfg *Config, container v1.Container, pod v1.Pod) (string, error) {
+	var capacity string
+	for _, volumeMount := range container.VolumeMounts {
+		for _, volume := range pod.Spec.Volumes {
+			if volume.Name == volumeMount.Name && volume.PersistentVolumeClaim != nil {
+				pvc, err := cfg.k8sClient.CoreV1().PersistentVolumeClaims(cfg.namespace).Get(
+					ctx,
+					volume.PersistentVolumeClaim.ClaimName,
+					metav1.GetOptions{},
+				)
+
+				if err != nil {
+					return "", errors.Wrapf(
+						err,
+						"error getting PVC %s",
+						volume.PersistentVolumeClaim.ClaimName,
+					)
+				}
+
+				capacity = pvc.Status.Capacity.Storage().String()
+				break
+			}
+		}
+	}
+	return capacity, nil
+}
+
+// DockerClientInterface defines the interface for interacting with the Docker API.
+type DockerClientInterface interface {
```

Comment:
I don't think this needs to be here if we break some of the code up and change how the tests work. Lets pair on this though since it's a bit more than a PR suggestion 💪

Reply:
Sounds good to me. I'll put some time on the calendar.